src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -707,10 +707,81 @@ def algebraic_dependence(self):
         RRR = PolynomialRing(K,r,tuple(Ts),order='negdeglex')
         return RRR.ideal(JJ.gens())
 
+    def coefficients_monomials(self, order=None, sparse=True):
+        """
+        Return tuple ``(A,v)`` where ``A`` is the coefficient matrix
```

Comment:
Return *the* tuple...

In fact, maybe:

Return the matrix of coefficients ``A`` and the vector of monomials ``v``, such that ``A v == self``.

Reply:
I think it is returning a tuple only.

```python
sage: F.coefficients_monomials()
(
[  0   0   0   0   0   0   0   0   0   1   2   2   2 126]
[  1   0   2   0   0   2   0   0   2 126   0   0   0   0]
[  0   2   0   0   2   0   0   2   0   0 126   0   0   0]
[  0   0   1   2   0   0   2   0   0   0   0 126   0   0],

(a^2, a*b, b^2, a*c, b*c, c^2, b*d, c*d, d^2, a, b, c, d, 1)
)
sage: type(F.coefficients_monomials())
<class 'tuple'>
```
> Return the matrix of coefficients A and the vector of monomials v, such that $A * v == self$

And, I am not able to figure out how I can do this because A * v and self belong to different classes, so I think the == operator will return False (Although I am not sure about it, but I think operator overloading can be done for this, but I am not able to find a way to do so). Do you have any idea how this can be done?, e.g., 

```python
sage: (F.coefficient_matrix()[0]*F.coefficient_matrix()[1]).__class__
<class 'sage.matrix.matrix_mpolynomial_dense.Matrix_mpolynomial_dense'>
sage: F.__class__
<class 'sage.rings.polynomial.multi_polynomial_sequence.PolynomialSequence_generic'>
```

Reply:
Yes, to be more precise the sentence should probably read

> Return the matrix of coefficients ``A`` and the vector of monomials ``v``, such that ``A v == vector(self)``.

("Return x and y, such that..." is saying, I think, the same as "Return the tuple ``x, y`` where x is ... and y is ..., such that ...", but it is shorter and easier to read, at least for me.)

I am not sure what is better.

Reply:
Oh... I thought you were saying that it is not returning a tuple. I am really sorry for the misunderstanding.

I have also updated this description. However, I have not changed this in the coefficient_matrix function because I was not sure whether I should change it or not.